dask/dataframe/core.py

```diff
@@ -2630,12 +2630,21 @@ def _is_single_last_div(x):
 
 def repartition_npartitions(df, npartitions):
     """ Repartition dataframe to a smaller number of partitions """
-    npartitions = min(npartitions, df.npartitions)
-    k = int(math.ceil(df.npartitions / npartitions))
-    divisions = df.divisions[::k]
-    if len(divisions) <= npartitions:
-        divisions = divisions + (df.divisions[-1],)
-    return df.repartition(divisions=divisions)
+    size = df.npartitions / npartitions
+    L = [int(i * size) for i in range(npartitions + 1)]
+    name = 'repartition-%d-%s' % (npartitions, df._name)
+    dsk = dict(
+        (
+            (name, i),
+            (pd.concat, [(df._name, j) for j in range(L[i], L[i + 1])])
+        ) for i in range(npartitions)
+    )
```

Comment:
Which "standard style" do you follow? Where can I read the rules? I follow [PEP-8](https://www.python.org/dev/peps/pep-0008/#indentation). Even though `pep8` and `pylint` throw a great number of style errors on Dask project, they find my changes almost fine (`pylint` is not happy about short variable names (`L`, `df`) and accessing private attribute `df._name`).